File: PGProblemSolving/기초문제/K번째수.cpp
//
//  K번째수.cpp
//  ProgrammersAlgorithmPractice
//
//  Created by MinKyeongTae on 13/09/2019.
//  Copyright © 2019 Min Kyeong Tae. All rights reserved.
//

// MARK: K번째 수
// MARK: array 특정 구간의(commands[i][0]~commands[i][1]번째 숫자) commands[i][2]번째로 큰 숫자를 구해라!

// MARK: - K번째수 복습 풀이답안
#if 0
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> Ans;
    for(auto &V: commands) {
        vector<int> TV = vector<int>(array.begin()+V[0]-1, array.begin()+V[1]);
        sort(TV.begin(), TV.end());
        
        Ans.push_back(TV[V[2]-1]);
    }
    
    return Ans;
}
#endif

//#include <string>
//#include <vector>
//#include <algorithm>
//
//using namespace std;
//
//int subVector(vector<int> v, const int &s, const int const &e, const int &o) {
//    vector<int> tv;
//    for(int i=s-1; i<=e-1; i++) {
//        tv.push_back(v[i]);
//    }
//    sort(tv.begin(), tv.end());
//    return tv[o-1];
//}
//
//vector<int> getKorderNumber(vector<int> array, vector<vector<int>> commands) {
//    vector<int> Ans;
//    for(int i=0; i<commands.size(); i++) {
//        Ans.push_back(subVector(array, commands[i][0], commands[i][1], commands[i][2]));
//    }
//    return Ans;
//}

File: PGProblemSolving/기초문제/K번째수.swift
//
//  K번째수.swift
//  ProgrammersAlgorithmPractice
//
//  Created by MinKyeongTae on 02/09/2019.
//  Copyright © 2019 Min Kyeong Tae. All rights reserved.
//

// MARK: - K번째 수

import Foundation

/// * Map, $ 인자값을 사용하여 간결하게 구현할 수 있다.
func getKorderNumbers(_ array: [Int], _ commands: [[Int]]) -> [Int] {
    return commands.map {
        Array(array[($0[0] - 1) ... ($0[1] - 1)]).sorted()[$0[2] - 1]
    }
}

// func solution(_ array:[Int], _ commands:[[Int]]) -> [Int] {
//    var ansArr = [Int]()
//    for (_,command) in commands.enumerated() {
//        var subArr = [Int]()
//        subArr = Array(array[(command[0]-1) ... (command[1]-1)])
//        ansArr.append(subArr.sorted()[command[2]-1])
//    }
//    return ansArr
// }
